hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/TransformStatefulP.java

```diff
@@ -93,6 +96,11 @@ protected boolean tryProcess(int ordinal, @Nonnull Object item) {
         return flatMapper.tryProcess((T) item);
     }
 
+    @Override
+    public void registerMetrics(MetricsContext context) {
+        UserMetricsUtil.cast(statefulFlatMapFn).registerMetrics(context);
```

Comment:
Would be better if all functions in which we support user metrics would be registered here. We rely on the code in `Processors.mapStatefulP()` that registers metrics for both the `createFn` and `statefulFlatMapFn` in `statefulFlatMapFn`.

Also, we should support metrics in `onEvictFn`.